src/CommandLineUtils/MissingParameterlessConstructorException.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace McMaster.Extensions.CommandLineUtils
+{
+    /// <summary>
+    /// The exception that is thrown when trying to intianciate a model with no parameterless constructor.
+    /// </summary>
+    /// <typeparam name="TModel">The type of the model to instanciate</typeparam>
+    public class MissingParameterlessConstructorException<TModel> : Exception
```

Comment:
To make this easier to use in a `catch` block, let's make this exception non-generic and make the type a property on the exception.

```suggestion
    public class MissingParameterlessConstructorException : Exception
    {
    	public Type Type { get; private set; }
```